### ISSUE TITLE
[BUG] download/export of my data not sending email with link

### ISSUE THREAD

**Issue by jaytat0** [RocketChat/Rocket.Chat#11914]
### Description:

Neither DOWNLOAD MY DATA nor EXPORT MY DATA under MY DATA in user's account/preferences are sending the email with link to download data file.

### Steps to reproduce:

1. go to MY DATA section of account/preferences
2. click DOWNLOAD MY DATA or EXPORT MY DATA
3. at "Your data file will be generated. A link to download it will be sent to your email address when ready." click send.

### Expected behavior:

email arrives with the link to download data file

### Actual behavior:

no email arrives with the link to download data file

### Server Setup Information:

Version of Rocket.Chat Server: 0.69.0 & repros on open.rocket.chat

**Comment by localguru** [comment]
Same here, no email arrives. EMail configuration itself is working (I get notification emails).

Logfiles on log level 2:

Download my data
```
Sep 25 07:28:31 testrocket testrocket[4690]: rocketchat_logger rocketchat_logger.js:278 #033[34mMeteor ➔ method#033[39m requestDataDownload -> userId: zTjbWHia5zCkPqF8f , arguments:  [ { fullExport: false } ]
```
export my data
```
Sep 25 07:28:59 testrocket testrocket[4690]: rocketchat_logger rocketchat_logger.js:278 #033[34mMeteor ➔ method#033[39m requestDataDownload -> userId: zTjbWHia5zCkPqF8f , arguments:  [ { fullExport: true } ]
```

### Server Setup Information:

- Version of Rocket.Chat Server: 0.69.2
- Operating System: Ubuntu 16.04 LTS
- Deployment Method: bundle self build
- Number of Running Instances: 1
- DB Replicaset Oplog: yes
- NodeJS Version: v8.11.3
- MongoDB Version: 3.4.17

**Comment by localguru** [comment]
Update: email was set 4 hours later, okay, but the I can't download the file: file not found.

The URL looks like this:

`https://domain.de/ufs/GridFS:UserDataFiles/k5oD6yhBrbFdBaw4g/2018-09-25-Local%2520Admin.zip`

RC is running behind a Nginx reverse proxy.

**Comment by jthomae1** [comment]
+1

No email was send. Even not after a couple of hours.

**Server Setup Information:**

- Version of Rocket.Chat Server: 0.69.2
- Operating System: Ubuntu 14.04.5 LTS
- Number of Running Instances: 1
- NodeJS Version: v8.11.3
- MongoDB Version: 3.4

**Comment by sampaiodiego** [comment]
@localguru just to make sure, were you using `GridFS` as your storage provider?

also, this process is indeed done in a very pace way, to not harm the server.. it's composed in many different steps, done one-by-one on 15 minutes (by default) interval and only 100 messages (by default) batches..

**Comment by localguru** [comment]
@sampaiodiego Diego, I'm running RC from tar ball, all data is located in local mongodb. Both RC and mongo are running in the same box.

**Comment by sampaiodiego** [comment]
thanks @localguru but what I meant was the setting on the Admin panel under `File Upload` section, what is the value for `Storage Type`? I'm asking this because the link you sent is for a `GridFS` storage type, so I wanted to make sure the link was generated correctly.

**Comment by localguru** [comment]
@sampaiodiego I understand. These are my settings:

![bildschirmfoto_2018-11-01_22-06-48](https://user-images.githubusercontent.com/8544984/47880169-d050a300-de22-11e8-9dce-e2270dda082d.png)

**Comment by protist** [comment]
My storage type is FileSystem, but I'm not getting the email either. All other emails are working.

If I set `Administration` > `User Data Download` > `System Path (Exported Files)` and `System Path (Compressed File)` to, say, `/tmp`, then should I be able to get the files from there, even if I don't receive the email notification? I don't see anything obvious in `/tmp`, so presumably the initial export stage isn't working either.

I do see the following directory structure, but apart from the sockets, it's only empty directories.
```
$ sudo tree /tmp/snap.rocketchat-server
/tmp/snap.rocketchat-server
└── tmp
    ├── mongodb-27017.sock
    ├── rocketchat-importer
    ├── ufs
    └── wHq2mJ9qY7obmGmvH
        ├── full
        │   └── assets
        └── partial
            └── assets

8 directories, 1 file
```

My `File Upload` location at  `/var/snap/rocketchat-server/common/upload` also contains nothing relevant.

**Comment by geekgonecrazy** [comment]
Does this still happen? Do you see any logs to indicate the export has actually ever finished?

**Comment by sampaiodiego** [comment]
the process can take a long time to export like many thousands of messages as it is done in chunks, so keep that in mind.

**Comment by protist** [comment]
> the process can take a long time to export like many thousands of messages as it is done in chunks, so keep that in mind.

I tested exporting almost immediately after installing Rocket.chat, with only ~6 messages present, and it still failed.

**Comment by protist** [comment]
Mandatory spam to feed the bot. (I replied almost immediately, so the tag(s) can be removed.)

**Comment by piotrkochan** [comment]
@sampaiodiego @geekgonecrazy don't you care about gdpr in your project? it is 2 years since this is not working

**Comment by protist** [comment]
More spam to feed the bot 😕

**Comment by sampaiodiego** [comment]
sorry about that @protist .. but we really need more details on this to be able to work on a fix..

it definitely works for me in a clean installation when I try it. 

maybe you can give full details of your setup and the actions you make to test the feature so we can to reproduce? like all the commands you do to install the server application, then describe the actions you do on the UI to start the export, then feed us with as most logs as you can.

**Comment by jthomae1** [comment]
I can only repeat my comment from 1. October 2018:

No email was send. Even not after a couple of hours. On three different servers!
In between I testet it with a lot of earlier versions. The export never worked.

Server Setup Information:

Version of Rocket.Chat Server: 3.7.1
Operating System: Ubuntu 16.04.7 LTS
Number of Running Instances: 1 (4 and 18 on 2 other servers)
NodeJS Version: v12.18.4
MongoDB Version: 4.0.13 (3.6.14 and 3.6.16 on 2 other servers)

**Comment by protist** [comment]
> sorry about that @protist .. but we really need more details on this to be able to work on a fix..
> 
> it definitely works for me in a clean installation when I try it.
> 
> maybe you can give full details of your setup and the actions you make to test the feature so we can to reproduce? like all the commands you do to install the server application, then describe the actions you do on the UI to start the export, then feed us with as most logs as you can.

@sampaiodiego I realised that because I'm running a 32-bit server, I'm stuck on an old version of Rocket.chat (1.3.2) so I'm not sure how useful troubleshooting will be. Nevertheless, it was a pretty standard install via snap. I can give you more details and logs if my version isn't too old for you to comment on.

**Comment by sampaiodiego** [comment]
hi @protist .. version 1.3.2 is definitely too old now, a lot of work have been done on this feature since then.

I didn't know we had different version for 32 and 64 bits of the snaps.. is that still a thing @LuluGO @geekgonecrazy ? @protist can you maybe try using a different snap channel? see https://forums.rocket.chat/t/introducing-snap-tracks/5890 
if doing anything with sensitive data, please don't forget to backup https://docs.rocket.chat/installation/snaps#how-do-i-backup-my-snap-data

**Comment by protist** [comment]
> I didn't know we had different version for 32 and 64 bits of the snaps.. is that still a thing

Yep, unfortunately 32-bit is stuck at 1.3.2: https://github.com/RocketChat/Rocket.Chat.Embedded.armhf

**Comment by sampaiodiego** [comment]
oh, you're running on arm 🤔  @Sing-Li is there anything you help us here?

**Comment by toowboga** [comment]
The issue still exists as I mentioned in my last comment.

**Comment by jthomae1** [comment]
The issue still exists as I mentioned in my last comment

**Comment by geekgonecrazy** [comment]
Yeah the issue isn't actually Rocket.Chat on 32bit.  Mongo dropped support for 32bit architecture and no longer build 32bit copies.

So while on 32 bit you are stuck on 1.3.2 the next version of Rocket.Chat drops support for the last working version of mongo for 32bit.  This is for a couple of reasons.  We can't support versions of mongo that mongo doesn't.  Also we started using features only in the newer versions of mongo.


How many of you experience this on latest version of Rocket.Chat?  3.8.1?  If so do you have installation method and steps to reproduce?

**Comment by jthomae1** [comment]
I will test it on 3.8.2 soon. We are using docker-compose setups on all three systems. But they are somewhat different from each other. The simplest has only one RC instance with a mongo db container.

**Comment by jthomae1** [comment]
Please reopen this issue, we are testing version 3.8.2 now and the issue still exists.

**Comment by sampaiodiego** [comment]
I've reopened and remove the labels causing it to be automatically closed.

@jthomae1 please give us more details as soon as you have. thx

**Comment by sennewood** [comment]
Still a problem in 3.15.0

**Comment by b90g** [comment]
7.2 , still not gdpr compliant.

**Comment by benedikt-wegmann** [comment]
Still a continued issue here. We don't have GridFS but file system storage backend.

Most of the time the process is "stuck" before generating the email to users. Users are also not offered a download link when revisiting the "my data" function.

I see the jobs in the MongoDB and can often find the corresponding zip files but the process don't complete, not even after days.
And since users often try again after a while this fills up our storage as well as some of these exports can be quite large.
